src/features/tutor/tutor.service.js

```diff
@@ -27,25 +28,70 @@ exports.getTutors = async ({
   approvalStatus = "approved",
   profileVisibility = "active",
   subjects,
-  availability,
+  name,
+  ratings,
   limit = 10,
   page = 1,
 }) => {
+  const includes = [];
+  let where = {
+    approvalStatus,
+    profileVisibility,
+  };
+
+  //Subject
   const subjectInclude = {
-    model: Subject,
+    model: Subject.scope("join"),
     as: "subjects",
   };
   if (subjects && subjects.length > 0) {
-    subjectInclude.query = {
-      [Op.in]: subjects,
+    subjectInclude.where = {
+      id: { [Op.in]: subjects.map(Number) },
+    };
+  }
+
+  includes.push(subjectInclude);
+
+  //Name
+  if (name) {
+    const searchWords = name.split(" ");
+
+    const conditions = searchWords.map((word) => ({
+      [Op.or]: [
+        { first_name: { [Op.iLike]: `%${word}%` } },
+        { last_name: { [Op.iLike]: `%${word}%` } },
+      ],
+    }));
+
+    const nameInclude = {
+      model: User.scope("join"),
+      as: "user",
+      where: conditions,
     };
+    includes.push(nameInclude);
+  }
+
+  //Ratings
+  //   const where = {
+  //   status: 'active',
+  //   funding: 'funded',
+  // };
+
+  // await sequelize.query(sql`SELECT * FROM projects WHERE ${sql.where(where)}`);
+  if (ratings && ratings.length > 0) {
+    const ratingWhere = sequelize.where(
+      sequelize.fn("ROUND", sequelize.col("rating")),
+      {
+        [Op.in]: ratings,
+      }
+    );
+    where = { [Op.and]: [{profileVisibility}, {approvalStatus}, ratingWhere] };
+    console.log(ratingWhere);
   }
```

Comment:
In the ratings branch we’re reassigning where, which drops any extra tutor-level conditions we might add later (currently it’s just approvalStatus/profileVisibility, but could be error prone in the future)

```suggestion
// Convert ratings safely:
const numericRatings = ratings.map(Number).filter((r) => !isNaN(r));

if (numericRatings.length) {
  const ratingWhere = sequelize.where(
    sequelize.fn("ROUND", sequelize.col("rating")),
    { [Op.in]: numericRatings }
  );
  where = { ...where, [Op.and]: [ratingWhere] };
}
```
This way the base where is preserved, and ratings get applied on top.


Also right now the ratings filter uses ROUND(rating) [Op.in] ratings, which means it only matches exact rounded values. e.g. 4.7 won’t match [4] because it rounds up to 5.
Did you mean this to work as exact stars only, or should it be a “minimum rating” filter (e.g. Op.gte)?

```
if (ratings && ratings.length > 0) {
  const minRating = Math.min(...ratings); // lowest value from the filter
  where = {
    ...where,
    rating: { [Op.gte]: minRating }, // tutors with rating >= minRating
  };
}
```

Reply:
The where stuff didn't exactly work but I've found a solution

Reply:
For the rating. I just felt tutors should fall into the category that is closer to their actual rating. 4.7 has tried he should be part of 5.0 people, 4.2 should be part of 4.0 people etc. We could do floor or ceiling but that seems a bit unfair.

Reply:
Well again that's actually a ux issue so I'm just going to leave it as is 😂